modules/editorial-metadata/lib/components/editorial-metadata-manager.js

```diff
@@ -0,0 +1,181 @@
+	const [ isModalVisible, setIsModalVisible ] = useState( false );
+
+	const handleErrorThrown = error => {
+		setSuccess( null );
+		setError( error );
+		setIsModalVisible( false );
+		setIsConfirmingDelete( false );
+		setEMetadataTerm( null );
+	};
+
+	const handleSuccess = ( message, eMetadataTermResult ) => {
+		setError( null );
+		setSuccess( message );
+		if ( eMetadataTerm && ! isConfirmingDelete ) {
+			setEMetadataTerms(
+				eMetadataTerms.map( eMetadataTerm => {
+					if ( eMetadataTerm.term_id === eMetadataTermResult.term_id ) {
+						return eMetadataTermResult;
+					}
+					return eMetadataTerm;
+				} )
+			);
+		} else if ( isConfirmingDelete ) {
+			setEMetadataTerms(
+				eMetadataTerms.filter(
+					eMetadataTerm => eMetadataTerm.term_id !== eMetadataTermResult.term_id
+				)
+			);
+		} else {
+			setEMetadataTerms( [ ...eMetadataTerms, eMetadataTermResult ] );
+		}
+		setIsModalVisible( false );
+		setIsConfirmingDelete( false );
+	};
+
+	const handleDelete = async () => {
+		try {
+			await apiFetch( {
+				url: VW_EDITORIAL_METADATA_CONFIGURE.url_edit_editorial_metadata + eMetadataTerm.term_id,
+				method: 'DELETE',
+			} );
+
+			handleSuccess(
+				sprintf(
+					__( 'Editorial Metadata "%s" deleted successfully.', 'vip-workflow' ),
+					eMetadataTerm.name
+				),
+				eMetadataTerm
+			);
+		} catch ( error ) {
+			handleErrorThrown( error.message );
+		}
+	};
+
+	const deleteModal = (
+		<ConfirmDeleteModal
+			confirmationMessage={ '' }
+			dataType={ 'metadata field' }
+			name={ eMetadataTerm?.name }
+			onCancel={ () => setIsConfirmingDelete( false ) }
+			onConfirmDelete={ handleDelete }
+		/>
+	);
+
+	const createEditModal = (
+		<CreateEditEditorialMetadataModal
+			availableMetadataTypes={ availableMetadataTypes.map( availableMetadataType => {
+				return {
+					value: availableMetadataType,
+					label: availableMetadataType,
+				};
+			} ) }
+			metadata={ eMetadataTerm }
+			onCancel={ () => setIsModalVisible( false ) }
+			onSuccess={ handleSuccess }
+			onErrorThrown={ handleErrorThrown }
+		/>
+	);
+
+	return (
+		<>
+			{ <SuccessNotice success={ success } /> }
+			{ error && (
+				<div style={ { marginBottom: '1rem' } }>
+					<Notice status="error" isDismissible={ true } onRemove={ () => setError( null ) }>
+						<p>{ error }</p>
+					</Notice>
+				</div>
+			) }
+			<Flex direction={ [ 'column' ] } justify={ 'start' } align={ 'start' }>
+				<FlexItem>
+					<Button
+						variant="secondary"
+						onClick={ () => {
+							setEMetadataTerm( null );
+							setIsModalVisible( true );
+						} }
+					>
+						{ __( 'Add New Metadata', 'vip-workflow' ) }
+					</Button>
+				</FlexItem>
+				<Flex className="emetadata-items" direction={ [ 'column', 'row' ] } justify={ 'start' }>
+					{ eMetadataTerms.map( eMetadataTerm => {
+						return (
+							<FlexItem className="emetadata-item" key={ eMetadataTerm.term_id }>
+								<Card>
```

Comment:
Do you think you can move this `<Card>` into a new component? Or I'd like other ways we may be able to reduce the complexity of this component split part of it somewhere else.

Reply:
I get what you mean, as there is a lot going on here. Lemme try to attempt that and see what I can do

Reply:
This made me realize that the error notice during an edit, and add operation should be raised within the modal dialog itself. It shouldn't go on the main page, as then you need to re-enter all your info. Besides that, I tried to simplify this down a touch by making a common error notice. 

I tried to move the `<Card>` to a new component, but in order to get it working the complexity is nearly the same. This is because the API calls need to be made, and the resulting state still needs to be managed. When we switch the custom status to be modal dialog based, I suspect there will be some common components made and that point we could reduce this down a touch. I'll leave this as is for now.